Microsoft.Toolkit.Uwp.UI.Controls/Carousel/CarouselItemAutomationPeer.cs

```diff
@@ -0,0 +1,217 @@
+            if (!string.IsNullOrEmpty(name))
+            {
+                return name;
+            }
+
+            return string.Empty;
+        }
+
+        /// <summary>
+        /// Called by GetAutomationId that gets the **AutomationId** of the element that is associated with the automation peer.
+        /// </summary>
+        /// <returns>
+        /// The string that contains the automation ID.
+        /// </returns>
+        protected override string GetAutomationIdCore()
+        {
+            var automationId = base.GetAutomationIdCore();
+            if (!string.IsNullOrEmpty(automationId))
+            {
+                return automationId;
+            }
+
+            if (this.OwnerCarouselItem != null)
+            {
+                return this.GetNameCore();
+            }
+
+            return string.Empty;
+        }
+
+        /// <summary>
+        /// Gets the control pattern that is associated with the specified Windows.UI.Xaml.Automation.Peers.PatternInterface.
+        /// </summary>
+        /// <param name="patternInterface">A value from the Windows.UI.Xaml.Automation.Peers.PatternInterface enumeration.</param>
+        /// <returns>The object that supports the specified pattern, or null if unsupported.</returns>
+        protected override object GetPatternCore(PatternInterface patternInterface)
+        {
+            switch (patternInterface)
+            {
+                case PatternInterface.SelectionItem:
+                    return this;
+            }
+
+            return base.GetPatternCore(patternInterface);
+        }
+
+        /// <summary>
+        /// Returns the size of the set where the element that is associated with the automation peer is located.
+        /// </summary>
+        /// <returns>
+        /// The size of the set.
+        /// </returns>
+        protected override int GetSizeOfSetCore()
+        {
+            int sizeOfSet = base.GetSizeOfSetCore();
+
+            if (sizeOfSet != -1)
+            {
+                return sizeOfSet;
+            }
+
+            CarouselItem owner = this.OwnerCarouselItem;
+            Carousel parent = owner.ParentCarousel;
+            sizeOfSet = parent.Items.Count;
+
+            return sizeOfSet;
+        }
+
+        /// <summary>
+        /// Returns the ordinal position in the set for the element that is associated with the automation peer.
+        /// </summary>
+        /// <returns>
+        /// The ordinal position in the set.
+        /// </returns>
+        protected override int GetPositionInSetCore()
+        {
+            int positionInSet = base.GetPositionInSetCore();
+
+            if (positionInSet != -1)
+            {
+                return positionInSet;
+            }
+
+            CarouselItem owner = this.OwnerCarouselItem;
+            Carousel parent = owner.ParentCarousel;
+            positionInSet = parent.IndexFromContainer(owner);
+
+            return positionInSet;
```

Comment:
Is IndexFromContainer 0-based or 1-based? According to [documentation](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.peers.automationpeer.getpositioninset?view=winrt-19041#Windows_UI_Xaml_Automation_Peers_AutomationPeer_GetPositionInSet) `PositionInSet` should be 1-based.

Reply:
Well spotted. I even read the documentation for this and completely missed that.

Reply:
Should be sorted now